File: house-robber/house-robber.cs
public class Solution {
    public int Rob(int[] nums) {
        
        int prev =0;
            int curr =0;
        foreach(int i in nums)
        {
            int temp = curr;
            curr = Math.Max(prev+i,curr);
            prev = temp;
            Console.WriteLine(temp);
            Console.WriteLine(curr);
            Console.WriteLine(prev);
        }
        return curr;
        
    }
}

File: solutions/HouseRobber.java

public class HouseRobber {
	 public int rob(int[] nums) {
	        
	        int taken=0;
	        int nottaken=0;
	       
	    
	       int max=Integer.MIN_VALUE;
	        
	for( int i=0;i<nums.length;i++){
	    
	   int  temp= taken;
	     taken=nottaken+nums[i];
	     nottaken=Math.max(temp,nottaken);
	    max=Math.max(max,Math.max(taken,nottaken));

	}    
	        return max;
	    
	        
	        
	    }
}
